sdk/storage/azblob/zc_leaseClient.go

```diff
@@ -0,0 +1,168 @@
+package azblob
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"net/url"
+	"time"
+)
+
+type BlobLeaseClient struct {
+	BlobClient
+	leaseId      *string
+	lastModified *time.Time
+	eTag         *string
+}
+
+func NewBlobLeaseClient(blobURL string, cred azcore.Credential, pathRenameMode *PathRenameMode, options *connectionOptions, leaseId *string) (BlobLeaseClient, error) {
+	u, err := url.Parse(blobURL)
+	if err != nil {
+		return BlobLeaseClient{}, err
+	}
+	con := newConnection(blobURL, cred, options)
+	blobClient := BlobClient{client: &blobClient{con, pathRenameMode}, u: *u}
+	return BlobLeaseClient{BlobClient: blobClient, leaseId: leaseId}, nil
+}
+
+// URL returns the URL endpoint used by the BlobLeaseClient object.
+func (blc BlobLeaseClient) URL() url.URL {
+	return blc.u
+}
+
+// String returns the URL as a string.
+func (blc BlobLeaseClient) String() string {
+	u := blc.URL()
+	return u.String()
+}
+
+// WithPipeline creates a new BlobLeaseClient object identical to the source but with the specified request policy pipeline.
+func (blc BlobLeaseClient) WithPipeline(pipeline azcore.Pipeline) BlobLeaseClient {
+	con := newConnectionWithPipeline(blc.u.String(), pipeline)
+	blobClient := BlobClient{client: &blobClient{con, blc.client.pathRenameMode}, u: blc.u}
+	return BlobLeaseClient{BlobClient: blobClient}
+}
+
+// AcquireLease acquires a lease on the blob for write and delete operations. The lease Duration must be between
+// 15 to 60 seconds, or infinite (-1).
+// For more information, see https://docs.microsoft.com/rest/api/storageservices/lease-blob.
+func (blc *BlobLeaseClient) AcquireLease(ctx context.Context, options *AcquireLeaseBlobOptions) (BlobAcquireLeaseResponse, error) {
```

Comment:
Why do we take AcquireLeaseOptions here and the same kind of lease options in the new lease client function?

Reply:
(though this question should probably be facing the opposite way-- why do we provide those options in the constructor?)